node/index.js

```diff
@@ -731,42 +731,50 @@ TChannelConnection.prototype.request = function request(options) {
     return req;
 };
 
-TChannelConnection.prototype.runInOp = function runInOp(handler, req, res) {
+TChannelConnection.prototype.handleRequest = function handleRequest(req) {
     var self = this;
+    req.remoteAddr = self.remoteName;
+    var res = self.handler.buildOutgoingResponse(req);
     var id = req.id;
     self.inPending++;
     var op = self.inOps[id] = new TChannelServerOp(self,
-        handler, self.channel.now(), req, opDone);
+        self.channel.now(), req, res);
+
+    var done = false;
+    process.nextTick(function runHandler() {
+        self.channel.handleRequest(req, res);
+    });
 
-    function opDone(err, res1, res2) {
+    res.on('end', opDone);
+    res.on('error', function onResError(err) {
```

Comment:
why do we handle error here ? feels like the wrong place to handle. `channel.handleRequest()` should really handle it